incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -33,18 +35,29 @@ var (
 
 var treeCmd = &cobra.Command{
 	Use:   "tree",
-	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.ExactArgs(1),
+	Short: "Displays the hierarchy tree rooted at the given namespaces separated by space and if no namespace is provided, displays the tree starting at every root namespace on the cluster.",
 	Run: func(cmd *cobra.Command, args []string) {
-		nnm := args[0]
+		defaultList := len(args) == 0
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		hier := getHierarchy(nnm)
-		fmt.Println(nameAndFootnotes(hier))
-		printSubtree("", hier)
-
+		var nsList []string
+		if defaultList {
+			nsList = getAllNamespaces()
```

Comment:
This seems different from most other k8s CLIs where if you don't provice a default one is chosen for you and you need to specify `--all-namespaces` to get them all. This is to prevent unnecessary strain on the api server (I suspect). 

@adrianludwin thoughts?

Reply:
So you suggestion would be:

```
kubectl hnc tree foo # displays foo
kubectl hnc tree # error
kubectl hnc tree --all-namespaces # displays the tree of all namespaces
```

Is that correct?

Reply:
That seems pretty reasonable to me. @shivi28 can you please make this change?  See [here](https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/pkg/kubectl/set.go#L77) for an example of how to add flags.

Reply:
And see https://github.com/spf13/cobra#persistent-flags to show how to add short aliases (eg `-A` as a synonym for `--all-namespaces`)

Reply:
yaa @rjbez17 @adrianludwin I will do it